### PR TITLE
fix(expect): always show custom message

### PR DESCRIPTION
**Pull request by eryue0220**
This pr is to solve #6206

### CODE REVIEW

File: expect/_matchers.ts [hunk 1]
```diff
@@ -89,13 +89,15 @@ export function toBeCloseTo(
   if (context.isNot) {
     if (pass) {
       throw new AssertionError(
-        `Expected the value not to be close to ${expected} (using ${numDigits} digits), but it is`,
+        context.customMessage ??
+          `Expected the value not to be close to ${expected} (using ${numDigits} digits), but it is`,
```
Comment (kt3k) [inline expect/_matchers.ts:93]: I think this part should be something like:

```ts
const defaultMessage = `Expected the value not to be close to ${expected} (using ${numDigits} digits), but it is`
throw new AssertionError(
  context.customMessage ? `${context.customMessage}: ${defaultMessage}` : defaultMessage
);
```

Vitest, for example, shows the custom message like the above.

In Vitest:
```ts
expect(42, "custom message").toEqual(43);
```
This assertion error shows the message like:
```
AssertionError: custom message: expected 42 to deeply equal 43

- Expected
+ Received

- 43
+ 42
```

Comment (eryue0220) [reply expect/_matchers.ts]: Updated

Comment (kt3k) [reply expect/_matchers.ts]: Oh, I missed this message. Sorry for the delay!

**Comment by kt3k** [comment]
I vendored `assertIsError` to `expect/` because the suggested change was not desired for `@std/assert/is-error`.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**